lib/Doctrine/ORM/Internal/Hydration/SimpleObjectHydrator.php

```diff
@@ -134,6 +136,11 @@ protected function hydrateRowData(array $sqlResult, array &$result)
 
             // Prevent overwrite in case of inherit classes using same property name (See AbstractHydrator)
             if ( ! isset($data[$fieldName]) || ! $valueIsNull) {
+                // If we have inheritance in resultset, make sure the field belongs to the correct class
+                if (isset($cacheKeyInfo['discriminatorValues']) && ! in_array($discrColumnValue, $cacheKeyInfo['discriminatorValues'], true)) {
```

Comment:
Would it be possible to combine this with the previous if statement? It would simplify things, wouldn't it?

Reply:
No its not possible.